File: problems/26/solution.go
func removeDuplicates(nums []int) int {
    i, j := 0, 1
    
    for j < len(nums) {
        if nums[i] != nums[j] {
            i++
            nums[i] = nums[j]
        }
        
        j++
    }
    
    return i + 1
}

File: problems/26/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (empty(nums)) {
            return 0;
        }
        
        int j = 1;
        
        for (int i = 1; i < size(nums); i++) {
            nums[j] = nums[i];
            
            if (nums[j] != nums[j - 1]) {
                j++;
            }
        }
        
        return j;
    }
};